tests/test_sqlPreFlight.py

```diff
@@ -0,0 +1,240 @@
+                                        ('Sensor', 'sensor'),
+                                        ('Exposure', 'exposure'),
+                                        ('Visit', 'visit')))
+        rows = list(rows)
+        self.assertEqual(len(rows), 4*3)   # 4 exposures times 3 sensors
+
+        # limit to single visit
+        headers, rows = self.preFlight.selectDataUnits(collections=["test"],
+                                                       expr="Visit.visit = 10",
+                                                       neededDatasetTypes=[rawType],
+                                                       futureDatasetTypes=[calexpType])
+        self.assertEqual(len(headers), 4)  # headers must be the same
+        rows = list(rows)
+        self.assertEqual(len(rows), 2*3)   # 2 exposures times 3 sensors
+
+        # expression excludes everyhting
+        headers, rows = self.preFlight.selectDataUnits(collections=["test"],
+                                                       expr="Visit.visit = 10 and Sensor.sensor > 1",
+                                                       neededDatasetTypes=[rawType],
+                                                       futureDatasetTypes=[calexpType])
+        self.assertEqual(len(headers), 4)  # headers must be the same
+        rows = list(rows)
+        self.assertEqual(len(rows), 2*2)   # 2 exposures times 2 sensors
+
+        # expression excludes everyhting
+        headers, rows = self.preFlight.selectDataUnits(collections=["test"],
+                                                       expr="Visit.visit > 1000",
+                                                       neededDatasetTypes=[rawType],
+                                                       futureDatasetTypes=[calexpType])
+        self.assertEqual(len(headers), 4)  # headers must be the same
+        rows = list(rows)
+        self.assertEqual(len(rows), 0)
+
+        # Selecting by PhysicalFilter, this should be done via
+        # PhysicalFilter.physical_filter but that is not supported currently,
+        # instead we can do Visit.physical_filter (or Exposure.physical_filter)
+        headers, rows = self.preFlight.selectDataUnits(collections=["test"],
+                                                       expr="Visit.physical_filter = 'dummy_r'",
+                                                       neededDatasetTypes=[rawType],
+                                                       futureDatasetTypes=[calexpType])
+        self.assertEqual(len(headers), 4)  # headers must be the same
+        rows = list(rows)
+        self.assertEqual(len(rows), 2*3)   # 2 exposures times 3 sensors
+
+    def testPreFlightSkyMapUnits(self):
+        """Test involving only SkyMap units, no joins to Camera"""
+        registry = self.registry
+
+        # need a bunch of units and datasets for test, we want 'AbstractFilter'
+        # in the test so also have to add PhysicslPhilter units
```

Comment:
I think this workaround can also now be removed.

Reply:
Hmm, I think AbstractFilter is a view on PhysicalFilter in the new implementation too, so I need to have some PhysicalFilter units in the registry, no?

Reply:
That is correct: AbstractFilter is just a SELECT DISTINCT on PhysicalFilter.